signac/contrib/filterparse.py

```diff
@@ -91,14 +92,65 @@ def parse_filter_arg(args, file=sys.stderr):
         if _is_json(args[0]):
             return _parse_json(args[0])
         else:
-            return _with_message(_parse_simple(args[0]), file)
+            key, value = _parse_simple(args[0])
+            return _with_message({key: value}, file)
     else:
-        q = dict()
-        for i in range(0, len(args), 2):
-            key = args[i]
-            if i+1 < len(args):
-                value = args[i+1]
-            else:
-                value = None
-            q.update(_parse_simple(key, value))
+        q = dict(parse_simple(args))
+
         return _with_message(q, file)
+
+
+def parse_simple(tokens):
```

Comment:
I would prefer to keep this "private" by renaming to `_parse_simple` and renaming the current `_parse_simple` to `_parse_simple_single` or so.

Reply:
I'm ok with the name change, however, I was actually hoping to expose this function publicly, to make it easier to implement custom user scripts like this:
```python
import click

@click.command()
@click.argument('filter', nargs=-1)
def cli(filter):
    jobs = project.find_jobs(parse_simple(filter))

cli()
```

Reply:
I support making this a public method / official API - if it's not public then `signac-dashboard` has to rely on a private method.

Reply:
Got it, in that case I support making it part of the API as well. I still like renaming the `_parse_simple` method to indicate that it operates on one filter component at a time, but we don't have to change the `parse_simple` method.